packages/cacti-plugin-ledger-connector-stellar/src/main/typescript/plugin-ledger-connector-stellar.ts

```diff
@@ -129,10 +132,36 @@ export class PluginLedgerConnectorStellar
     return response;
   }
 
-  public transact(): Promise<unknown> {
-    throw new InternalServerError("Method not implemented.");
+  public async runSorobanTransaction(
+    req: RunSorobanTransactionRequest,
+  ): Promise<RunSorobanTransactionResponse> {
+    const invokeArgs = {
+      ...req,
+      methodArgs: req.methodArgs ?? {},
+      networkConfig: this.networkConfig,
+      fnLogPrefix: this.className,
+      txInvocation: convertApiTransactionInvocationToStellarPlus(
+        req.transactionInvocation,
+        this.networkConfig,
+      ),
+    };
+
+    const result = req.readOnly
+      ? await invokeContract<typeof invokeArgs.methodArgs>({
+          ...invokeArgs,
+          readOnly: true,
+        })
+      : await invokeContract<typeof invokeArgs.methodArgs>({
+          ...invokeArgs,
+          readOnly: false,
+        });
+
+    this.prometheusExporter.addCurrentTransaction();
+    return { result } as RunSorobanTransactionResponse;
   }
 
+  public transact = this.runSorobanTransaction; // Review the interface
```

Comment:
@fazzatti Please create a method called transact and then call the runSorobanTransaction method within that (e.g. make it a wrapper to runSorobanTransaction)

Reply:
Thanks @petermetz !

Did a very simple wrapping for now. In the future, once we add other flavors of transactions we can expand this wrapper to handle multiple options.

Reply:
@fazzatti Thank you, LGTM! :-)